### PR TITLE
Add/fix build capability for Gaea-C5, Gaea-C6, and container

### LINKED ISSUE #799: Add/fix build capability for Gaea-C5, Gaea-C6, and container
After the recent Gaea-C5 OS upgrade, gsi-enfk fails to build.
This issues corrects Gaea-C5 build, adds Gaea-C6 build capability (following ufs-wx-model [PR2448](https://github.com/ufs-community/ufs-weather-model/pull/2448)), and adds containerized build capability.

Refs NOAA-EMC/global-workflow [3011](https://github.com/NOAA-EMC/global-workflow/issues/3011)
Refs NOAA-EMC/global-workflow [3025](https://github.com/NOAA-EMC/global-workflow/issues/3025)

### PR DESCRIPTION
**Pull request by DavidBurrows-NCO**
Resolves #799 

**Type of change**
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
Cloned and built on Gaea-C5, Gaea-C6, and in a container.

**Checklist**

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: src/gsi/CMakeLists.txt [hunk 1]
```diff
@@ -155,7 +155,11 @@ target_link_libraries(gsi_fortran_obj PUBLIC nemsio::nemsio)
 target_link_libraries(gsi_fortran_obj PUBLIC ncio::ncio)
 target_link_libraries(gsi_fortran_obj PUBLIC w3emc::w3emc_d)
 target_link_libraries(gsi_fortran_obj PUBLIC sp::sp_d)
-target_link_libraries(gsi_fortran_obj PUBLIC bufr::bufr_d)
+if(DEFINED ENV{USE_BUFR4})
```
Comment (RussTreadon-NOAA) [inline src/gsi/CMakeLists.txt:158]: Looks OK to me.  Cross checking with @DavidHuber-NOAA .  PR #791 upgrades to bufr/12.1.0.  Not sure how the bufr logic added here might impact Dave's PR.

**Review by RussTreadon-NOAA — COMMENTED** [review]
Looks OK to me.  

Two comments:
1. I can't test the container.  
2. Additional changes are needed to run GSI/EnKF ctests on Gaea-C6.  @DavidBurrows-NCO , do you plan on adding these changes to this PR or will a new issue and PR be opened to activate GSI/EnKF ctests on Gaea-C6?

**Comment by DavidBurrows-NCO** [comment]
@RussTreadon-NOAA Thanks for taking a look
1) @mark-a-potts Do you want to say anything about the container buildling?
2) I have some time today, so let me look into the regression tests on C6. I'll get back to you today.

**Comment by DavidBurrows-NCO** [comment]
Hello @RussTreadon-NOAA. I've made good progress on GSI reg tests.  I'm currently using the same walltime/processor configuration as C5 for C6. This can be adjusted, but here are the current results::
```
Test project /gpfs/f6/bil-fire8/scratch/David.Burrows/oct24_gsi/c5c6con_branch/build
    Start 3: rrfs_3denvar_rdasens
    Start 1: global_4denvar
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
    Start 2: rtma
1/6 Test #2: rtma .............................   Passed  843.31 sec
2/6 Test #6: global_enkf ......................   Passed  844.64 sec
3/6 Test #5: hafs_3denvar_hybens ..............   Passed  1029.12 sec
4/6 Test #4: hafs_4denvar_glbens ..............   Passed  1154.60 sec
5/6 Test #1: global_4denvar ...................   Passed  1561.03 sec
```
rrfs_3denvar_rdasens_loproc_updat keeps hitting the wall clock even after I increased to 60 mins. It freezes in the same spot.  I've attached a text file of the output log.  I don't see anything too good in the working directory.  Please let me know your thoughts. Thanks!
[rrfs_output_wall_clock_C6_for_russ.odt](https://github.com/user-attachments/files/17634251/rrfs_output_wall_clock_C6_for_russ.odt)

**Comment by RussTreadon-NOAA** [comment]
Thank you @DavidBurrows-NCO for the update.  This looks good.  We've had problems with the `rrfs_3denvar_rdasens` test on other machines.  Tagging regional DA staff:  @TingLei-NOAA , @ShunLiu-NOAA , @hu5970

File: regression/regression_param.sh [hunk 2]
```diff
@@ -24,13 +24,28 @@ case $machine in
            memnode=96
            numcore=40
     ;;
+<<<<<<< HEAD
```
Comment (RussTreadon-NOAA) [inline regression/regression_param.sh]: Looks like an unresolved conflict.  I think we want to retain lines 28 to 39  and git rid of lines 27, 40 to 48.

**Comment by RussTreadon-NOAA** [comment]
**C6 ctests results**

@DavidBurrows-NCO , I obtained similar C6 ctest results
```
Test project /gpfs/f6/ira-sti/scratch/Russ.Treadon/git/gsi/pr800/build
    Start 2: rtma
    Start 1: global_4denvar
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #2: rtma .............................   Passed  851.24 sec
2/6 Test #6: global_enkf ......................   Passed  851.31 sec
3/6 Test #5: hafs_3denvar_hybens ..............   Passed  975.95 sec
4/6 Test #4: hafs_4denvar_glbens ..............   Passed  1275.93 sec
5/6 Test #1: global_4denvar ...................   Passed  1565.65 sec
```
`rrfs_3denvar_rdasens_loproc_updat` hung.  The job was killed when it reached the specified 30 minute wall clock limit.
```
gsi_metguess_mod*create_: alloc() for met-guess done
 guess_grids*create_chemges_grids: trouble getting number of chem/gases
 metvardeb333-2d name ps
 metvardeb333-2d name z
 metvardeb333-2d name t2m
 metvardeb333-2d name q2m
 metvardeb333-3d name u
 metvardeb333-3d name v
 metvardeb333-3d name w
 metvardeb333-3d name tv
 metvardeb333-3d name q
 metvardeb333-3d name oz
 metvardeb333-3d name delp
 metvardeb333-3d name ql
 metvardeb333-3d name qr
 metvardeb333-3d name qs
 metvardeb333-3d name qi
 metvardeb333-3d name qg
 metvardeb333-3d name dbz
 metvardeb333-3d name fed
  fv3lam_io_dynmetvars3d_nouv is wtsendelp
  fv3lam_io_tracermevars3d_nouv is qozqlqrqsqiqg
  fv3lam_io_phymetvars3d_nouv is dbzfed
 the metvarname z will be dealt separately
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
slurmstepd: error: *** STEP 207256804.0 ON c6n0992 CANCELLED AT 2024-11-05T13:18:35 DUE TO TIME LIMIT ***
```

**C5 ctest results**

Ctest behavior on C5 is very different.  The following tests ran and failed
```
Test project /gpfs/f5/nggps_emc/scratch/Russ.Treadon/git/gsi/pr800/build
    Start 2: rtma
    Start 1: global_4denvar
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #2: rtma .............................***Failed  484.22 sec
2/6 Test #6: global_enkf ......................***Failed  484.22 sec
3/6 Test #1: global_4denvar ...................***Failed  554.70 sec
```

_rtma_
The low task count rtma job failed at the end of the run.  Timing statistics are printed from task 0.  The traceback cites `mpi_finalize` in `/gpfs/f5/nggps_emc/scratch/Russ.Treadon/Russ.Treadon/gsi_tmp/ptmp/tmpreg_rtma/rtma_loproc_updat/stdout`
```
 GENSTATS_GPS:  no profiles to process (nprof_gfs=           0 ), EXIT routine
 gsi_metguess_mod*destroy_: dealloc() for met-guess done
 glbsoi: complete
[000]gsisub(): : complete.


     ENDING DATE-TIME    NOV 05,2024  12:28:09.735  310  TUE   2460620
     PROGRAM GSI_ANL HAS ENDED.
* . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * .
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source
libpthread-2.31.s  00001512DD61F910  Unknown               Unknown  Unknown
libmlx5.so.1.24.4  00001512D240FC4B  Unknown               Unknown  Unknown
libmlx5.so.1.24.4  00001512D245A81A  Unknown               Unknown  Unknown
libfabric.so.1.23  00001512D715FA73  Unknown               Unknown  Unknown
libfabric.so.1.23  00001512D7168CF7  Unknown               Unknown  Unknown
libfabric.so.1.23  00001512D717293C  Unknown               Unknown  Unknown
libfabric.so.1.23  00001512D71733EB  Unknown               Unknown  Unknown
libfabric.so.1.23  00001512D7175DD4  Unknown               Unknown  Unknown
libmpi_intel.so.1  00001512D9066FFE  Unknown               Unknown  Unknown
libmpi_intel.so.1  00001512D8EBA484  Unknown               Unknown  Unknown
libmpi_intel.so.1  00001512D7945854  MPI_Finalize          Unknown  Unknown
libmpifort_intel.  00001512DD6742B9  pmpi_finalize__       Unknown  Unknown
gsi.x              000000000041502F  MAIN__                    643  gsimain.f90

...

*****************RESOURCE STATISTICS*******************************
The total amount of wall time                        = 187.684936
The total amount of time in user mode                = 178.017540
The total amount of time in sys mode                 = 4.573581
The maximum resident set size (KB)                   = 1872772
Number of page faults without I/O activity           = 309201
Number of page faults with I/O activity              = 301
Number of times filesystem performed INPUT           = 2440
Number of times filesystem performed OUTPUT          = 0
Number of Voluntary Context Switches                 = 8665
Number of InVoluntary Context Switches               = 177
*****************END OF RESOURCE STATISTICS*************************

forrtl: severe (174): SIGSEGV, segmentation fault occurred
```

_global_enkf_
Here the low task count job ran to completion but the high task count job failed.  Again, task 0 wrote timing statistics to `stdout`, but `/gpfs/f5/nggps_emc/scratch/Russ.Treadon/Russ.Treadon/gsi_tmp/ptmp/tmpreg_global_enkf/global_enkf_hiproc_updat/stderr`  contains
```
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source
libpthread-2.31.s  0000152997AAD910  Unknown               Unknown  Unknown
libmlx5.so.1.24.4  000015298B0C8C4B  Unknown               Unknown  Unknown
libmlx5.so.1.24.4  000015298B11381A  Unknown               Unknown  Unknown
libfabric.so.1.23  0000152991661A73  Unknown               Unknown  Unknown
libfabric.so.1.23  000015299166ACF7  Unknown               Unknown  Unknown
libfabric.so.1.23  000015299167493C  Unknown               Unknown  Unknown
libfabric.so.1.23  00001529916753EB  Unknown               Unknown  Unknown
libfabric.so.1.23  0000152991677DD4  Unknown               Unknown  Unknown
libmpi_intel.so.1  0000152993568FFE  Unknown               Unknown  Unknown
libmpi_intel.so.1  00001529933BC484  Unknown               Unknown  Unknown
libmpi_intel.so.1  0000152991E47854  MPI_Finalize          Unknown  Unknown
libmpifort_intel.  000015299F28C2B9  pmpi_finalize__       Unknown  Unknown
enkf.x             000000000046C65A  mpisetup_mp_mpi_c         134  mpisetup.f90
enkf.x             0000000000414D68  MAIN__                    281  enkf_main.f90
enkf.x             000000000041423D  Unknown               Unknown  Unknown
libc-2.31.so       0000152996DF924D  __libc_start_main     Unknown  Unknown
enkf.x             000000000041416A  Unknown               Unknown  Unknown
```
Line 134 of `src/enkf/mpisetup.f90` is `call mpi_finalize(ierr)`

_global_4denvar_
The low task count job failed just like the low task count rtma.  Timing statistics from task 0 are printed to `/gpfs/f5/nggps_emc/scratch/Russ.Treadon/Russ.Treadon/gsi_tmp/ptmp/tmpreg_global_4denvar/global_4denvar_loproc_updat/stdout` along with a traceback pointing to `mpi_finalize`
```
 destroy_ges_derivatives: successfully complete
 destroy_ges_tendencies: successfully complete
 gsi_chemguess_mod*destroy_: dealloc() for chem-tracer done
 gsi_metguess_mod*destroy_: dealloc() for met-guess done
 glbsoi: complete
[000]gsisub(): : complete.


     ENDING DATE-TIME    NOV 05,2024  12:29:31.208  310  TUE   2460620
     PROGRAM GSI_ANL HAS ENDED.
* . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * . * .
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source
libpthread-2.31.s  0000148FF5C7C910  Unknown               Unknown  Unknown
libmlx5.so.1.24.4  0000148FEA890C4B  Unknown               Unknown  Unknown
libmlx5.so.1.24.4  0000148FEA8DB81A  Unknown               Unknown  Unknown
libfabric.so.1.23  0000148FEED5FA73  Unknown               Unknown  Unknown
libfabric.so.1.23  0000148FEED68CF7  Unknown               Unknown  Unknown
libfabric.so.1.23  0000148FEED7293C  Unknown               Unknown  Unknown
libfabric.so.1.23  0000148FEED733EB  Unknown               Unknown  Unknown
libfabric.so.1.23  0000148FEED75DD4  Unknown               Unknown  Unknown
libmpi_intel.so.1  0000148FF0C66FFE  Unknown               Unknown  Unknown
libmpi_intel.so.1  0000148FF0ABA484  Unknown               Unknown  Unknown
libmpi_intel.so.1  0000148FEF545854  MPI_Finalize          Unknown  Unknown
libmpifort_intel.  0000148FF5CD62B9  pmpi_finalize__       Unknown  Unknown
gsi.x              000000000041502F  MAIN__                    643  gsimain.f90
gsi.x              0000000000414FBD  Unknown               Unknown  Unknown

...

*****************RESOURCE STATISTICS*******************************
The total amount of wall time                        = 439.916953
The total amount of time in user mode                = 403.179300
The total amount of time in sys mode                 = 8.852160
The maximum resident set size (KB)                   = 1257204
Number of page faults without I/O activity           = 443816
Number of page faults with I/O activity              = 1279
Number of times filesystem performed INPUT           = 402022
Number of times filesystem performed OUTPUT          = 0
Number of Voluntary Context Switches                 = 38081
Number of InVoluntary Context Switches               = 300
*****************END OF RESOURCE STATISTICS*************************
```

Not sure what's going on here.  Do we need to (un)set certain environment variables on C5?

The remaining tests
```
hafs_3denvar_hybens
hafs_4denvar_glbens
rrfs_3denvar_rdasens
```
were all killed by the system after reaching the specified wall clock limit.  Interestingly, the low task count `hafs_3denvar` and `hafs_4denvar` jobs ran to completion.  The high task count jobs did not finish.  It looks like they hung while reading radar data.  
```
  for radar KDGX nsuper=          57  delazmmax=  0.248988632686689      T
  vrmin,max=  -12.1000000000000        9.08870967741935       errmin,max=
  0.000000000000000E+000   4.54726892432853
  deltiltmin,max=  3.818932603314051E-002  0.605471355033601
  deldistmin,max=  -387.290631756274       -5.25372147616690
slurmstepd: error: *** STEP 135229599.0 ON c5n0775 CANCELLED AT 2024-11-05T12:50:00 DUE TO TIME LIMIT ***
```

The C5 `rrfs_3denvar_rdasens` job hangs in the same way as observed on C6
```
 gsi_metguess_mod*create_: alloc() for met-guess done
 guess_grids*create_chemges_grids: trouble getting number of chem/gases
 metvardeb333-2d name ps
 metvardeb333-2d name z
 metvardeb333-2d name t2m
 metvardeb333-2d name q2m
 metvardeb333-3d name u
 metvardeb333-3d name v
 metvardeb333-3d name w
 metvardeb333-3d name tv
 metvardeb333-3d name q
 metvardeb333-3d name oz
 metvardeb333-3d name delp
 metvardeb333-3d name ql
 metvardeb333-3d name qr
 metvardeb333-3d name qs
 metvardeb333-3d name qi
 metvardeb333-3d name qg
 metvardeb333-3d name dbz
 metvardeb333-3d name fed
  fv3lam_io_dynmetvars3d_nouv is wtsendelp
  fv3lam_io_tracermevars3d_nouv is qozqlqrqsqiqg
  fv3lam_io_phymetvars3d_nouv is dbzfed
 the metvarname z will be dealt separately
slurmstepd: error: *** STEP 135229565.0 ON c5n0801 CANCELLED AT 2024-11-05T12:36:00 DUE TO TIME LIMIT ***
```

**Comment by CoryMartin-NOAA** [comment]
@RussTreadon-NOAA I vaguely recall something like this previously, like 6-9 months ago, where GSI would run but crash at the very end of execution. Do you recall this, or am I imagining it?

**Comment by RussTreadon-NOAA** [comment]
@CoryMartin-NOAA , this sounds vaguely familiar.  Let me check GSI issues and PRs for clues.  The rrfs failure is a known problem.

**Comment by DavidBurrows-NCO** [comment]
Hi @RussTreadon-NOAA. I know it's not the solution you want, but I adjusted the node/processor configuration to match Hera on C6 and rrfs was successful:
```
David.Burrows@gaea65 07:49 build $ ctest -j 6
Test project /gpfs/f6/bil-fire8/scratch/David.Burrows/oct24_gsi/c5c6con_branch/build
    Start 1: global_4denvar
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  487.91 sec
2/6 Test #6: global_enkf ......................   Passed  848.19 sec
3/6 Test #2: rtma .............................   Passed  902.67 sec
4/6 Test #5: hafs_3denvar_hybens ..............   Passed  965.08 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  1094.00 sec
6/6 Test #1: global_4denvar ...................   Passed  1561.08 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 1561.09 sec
```
I'm working on C5 right now.

**Comment by RussTreadon-NOAA** [comment]
@DavidBurrows-NCO :  Changing the task count is consistent with the regional DA team recommendation.  Thank you for looking at the C5 failures.

**Comment by RussTreadon-NOAA** [comment]
@TingLei-NOAA informed me that he will be on leave and is unable to update the task count for `rrfs_3denvar_rdasens` on various machines. 

@DavidBurrows-NCO  , please commit the modified `regression/regression_param.sh` you used to get a successful `rrfs_3denvar_rdasens` run on Gaea C6.   Thank you!

**Comment by RussTreadon-NOAA** [comment]
@DavidBurrows-NCO , we need two peer reviews for GSI PRs.  My review doesn't count as a peer review.  Who would you like to review this PR?

File: ush/sub_gaeac5 [hunk 3]
```diff
@@ -158,6 +158,7 @@ sbatch=${sbatch:-sbatch}
 ofile=$DATA/subout$$
 >$ofile
 chmod 777 $ofile
+export FI_VERBS_PREFER_XRC=0
```
Comment (RussTreadon-NOAA) [inline ush/sub_gaeac5]: Does this setting resolve what appears to be `mpi_finalize` problems on C5?

**Review by RussTreadon-NOAA — DISMISSED** [review]
Approve.

**Review by CoryMartin-NOAA — DISMISSED** [review]

File: ush/sub_gaeac5 (hunk 3 above)
Comment (DavidBurrows-NCO) [reply ush/sub_gaeac5]: > Does this setting resolve what appears to be `mpi_finalize` problems on C5?

It appears so.  Here is the notice from Seth Underwood with Gaea C5: "After the C5 update, users reported that some jobs failed during the MPI_Finalize call. We have alerted ORNL and HPE. HPE has suggested setting the environment variable FI_VERBS_PREFER_XRC=0 in the run script (setenv FI_VERBS_PREFER_XRC 0, for csh; export FI_VERBS_PREFER_XRC=0). This has resolved the error in our tests.  Please add this variable to your run script(s) if you also hit this error.  Please note that we do not see any issues preemptively setting this environment variable."  

Now that I think the MPI_Finalize issue is resolved, I am going to adjust the resources and test a little more. I'll let you know when I have my final changes in place for you to look over.

**Comment by RussTreadon-NOAA** [comment]
Excellent!  Thank you @DavidBurrows-NCO  for working through various issues.

**Comment by DavidBurrows-NCO** [comment]
@RussTreadon-NOAA Quick question...if a particular test fails, but I check all the stdout, and they return rc=0...does that typically mean the job took too long to run?  I assume there are set run time values for each test?  Thanks

**Comment by RussTreadon-NOAA** [comment]
Unfortunately, the checks in GSI ctests are not very robust.  Some of the timing and memory usage checks can yield false positives.  The test _Failed_ but a check of the results does not indicate a problem.  Since GSI has no code manager and we are transitioning to JEDI, it's unlikely the GSI ctests will be cleaned up to yield more consistent results.  If there's a particular failure you'd like me to look at, give me the path or the rundir and I'll take a  look.

**Review by CoryMartin-NOAA — DISMISSED** [review]

**Comment by RussTreadon-NOAA** [comment]
**Gaea C5 and C6 ctests**
Install `NOAA-EPIC:feature/c5c6conbuild` at 3b98cf49 on Gaea C5 and C6.  Run ctests with following results.  On both machines use `NOAA-EPIC:feature/c5c6conbuild` as both the _updat_ and _contrl_

_Gaea C5_
```
Test project /gpfs/f5/nggps_emc/scratch/Russ.Treadon/git/gsi/pr800/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #6: global_enkf ......................   Passed  1570.25 sec
2/6 Test #3: rrfs_3denvar_rdasens .............   Passed  1747.56 sec
3/6 Test #1: global_4denvar ...................   Passed  2688.64 sec
4/6 Test #4: hafs_4denvar_glbens ..............   Passed  3742.89 sec
5/6 Test #5: hafs_3denvar_hybens ..............   Passed  3743.19 sec
6/6 Test #2: rtma .............................***Failed  4817.39 sec

83% tests passed, 1 tests failed out of 6

Total Test time (real) = 4817.41 sec

The following tests FAILED:
          2 - rtma (Failed)
```
The `rtma` test failed due to
```
The runtime for rtma_loproc_updat is 252.267314 seconds.  This has exceeded maximum allowable threshold time of 201.010062 seconds, resulting in Failure time-thresh of the regression test.
```
Here are the `gsi.x` wall time for various runs of this test
```
rtma_hiproc_contrl/stdout:The total amount of wall time                        = 182.670720
rtma_hiproc_updat/stdout:The total amount of wall time                        = 180.407343
rtma_loproc_contrl/stdout:The total amount of wall time                        = 182.736420
rtma_loproc_updat/stdout:The total amount of wall time                        = 252.267314
```
Indeed the _loproc_updat_ wall time is considerably greater than the _loproc_contrl_ wall time.  Note, however, that the _updat_ and _contrl_ tests use the same `gsi.x`.  Thus, the wall time difference reflects system load, i/o speed, or other system related factors. This is not a fatal fail.

_Gaea C6_
```
Test project /gpfs/f6/ira-sti/scratch/Russ.Treadon/git/gsi/pr800/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  487.90 sec
2/6 Test #6: global_enkf ......................***Failed  905.26 sec
3/6 Test #2: rtma .............................***Failed  1144.95 sec
4/6 Test #4: hafs_4denvar_glbens ..............   Passed  1206.88 sec
5/6 Test #5: hafs_3denvar_hybens ..............   Passed  1454.39 sec
6/6 Test #1: global_4denvar ...................   Passed  1866.25 sec

67% tests passed, 2 tests failed out of 6

Total Test time (real) = 1866.26 sec

The following tests FAILED:
          2 - rtma (Failed)
          6 - global_enkf (Failed)
```
The `global_enkf` and `rtma` failures are for the same reason
```
The runtime for global_enkf_hiproc_updat is 238.742326 seconds.  This has exceeded maximum allowable threshold time of 209.509297 seconds, resulting in Failure timethresh2 of the regression test.
```

```
The runtime for rtma_hiproc_updat is 208.155676 seconds.  This has exceeded maximum allowable threshold time of 194.161564 seconds, resulting in Failure of timethresh2 the regression test.
```
Here are the wall times from the various tests
```
tmpreg_global_enkf/global_enkf_hiproc_contrl/stdout:The total amount of wall time                        = 190.462998
tmpreg_global_enkf/global_enkf_hiproc_updat/stdout:The total amount of wall time                        = 238.742326
tmpreg_global_enkf/global_enkf_loproc_contrl/stdout:The total amount of wall time                        = 159.689075
tmpreg_global_enkf/global_enkf_loproc_updat/stdout:The total amount of wall time                        = 147.498631
```

```
tmpreg_rtma/rtma_hiproc_contrl/stdout:The total amount of wall time                        = 176.510513
tmpreg_rtma/rtma_hiproc_updat/stdout:The total amount of wall time                        = 208.155676
tmpreg_rtma/rtma_loproc_contrl/stdout:The total amount of wall time                        = 171.044928
tmpreg_rtma/rtma_loproc_updat/stdout:The total amount of wall time                        = 167.905287
```
For both tests the _hiproc_updat_ wall time is notably greater than the _hiproc_contrl_.  This is interesting.  The _updat_ and _contrl_ runs use the same executables.  he wall time differences reflect differences in system load, i/o speed, or other aspects of the system.  This is not a fatal fail.

File: regression/regression_param.sh [hunk 4]
```diff
@@ -103,9 +111,12 @@ case $regtest in
...
            topts[1]="0:15:00" ; popts[1]="5/4/"  ; ropts[1]="/1"
            topts[2]="0:15:00" ; popts[2]="10/4/"  ; ropts[2]="/1"
-        elif [[ "$machine" = "Gaea" ]]; then
-           topts[1]="0:15:00" ; popts[1]="64/1/"  ; ropts[1]="/1"
-           topts[2]="0:15:00" ; popts[2]="128/2/"  ; ropts[2]="/1"
+        elif [[ "$machine" = "gaeac5" ]]; then
+           topts[1]="0:15:00" ; popts[1]="40/3/"  ; ropts[1]="/1"
+           topts[2]="0:15:00" ; popts[2]="40/5/"  ; ropts[2]="/1"
+        elif [[ "$machine" = "gaeac6" ]]; then
+           topts[1]="0:60:00" ; popts[1]="40/3/"  ; ropts[1]="/1"
```
Comment (RussTreadon-NOAA) [inline regression/regression_param.sh]: The C6 wall clock limit request on this and the following line can be reduced to 15 minutes.  A check of recent `rrfs_3denvar_rdasens` tests on C5 and C6 return `gsi.x` wall times ranging from 77 to 126 seconds.

**Review by RussTreadon-NOAA — COMMENTED** [review]
Looks good.  Please reduce the Gaea C6 wall clock limit for `rrfs_3denvar_rdasens` from `0:60:00` to `0:15:00`.

**Comment by RussTreadon-NOAA** [comment]
This PR is awaiting the return of WCOSS2 to developers so WCOSS2 ctests can be run.  Assuming reduction of the Gaea C6 `rrfs_3denvar_rdasens` wall time and acceptable WCOSS2 results, this PR can be merged into `develop`.

**Review by RussTreadon-NOAA — DISMISSED** [review]
Looks good to me.   Approve.

File: regression/regression_param.sh (hunk 4 above)
Comment (DavidBurrows-NCO) [reply regression/regression_param.sh]: Hi @RussTreadon-NOAA. I returned the wall clock for rrfs to 15 mins for C6.  I also moved the placement of `export FI_VERBS_PREFER_XRC=0` from `ush/sub_gaeac5` to `regression/regression_var.sh`. Let me know if that is not okay.  I reran the C5 regression tests successfully after making this change.  And I am getting similar behavior that you are.  The jobs are consistently completing successfully but with a bit of variability in run time.
When is WCOSS returned to the devs?

**Review by CoryMartin-NOAA — DISMISSED** [review]

File: regression/regression_param.sh (hunk 4 above)
Comment (RussTreadon-NOAA) [reply regression/regression_param.sh]: @DavidBurrows-NCO , we normally put system environment variables in machine specific sections at the end of `regression/regresson_param.sh`.   

As a test I moved FI_VERBS_PREFER_XRC=0 from `regression/regression_var.sh` to the `gaeac5` section at the end of `regression/regression_param.sh`
```
elif [[ "$machine" = "gaeac5" ]]; then
   export OMP_STACKSIZE=1024M
   export MPI_BUFS_PER_PROC=256
   export MPI_BUFS_PER_HOST=256
   export MPI_GROUP_MAX=256
   export FI_VERBS_PREFER_XRC=0
   export APRUN="srun --export=ALL -n \$ntasks"
elif [[ "$machine" = "gaeac6" ]]; then
```

I reran the ctests on Gaea C5.  All tests passed.
```
Test project /gpfs/f5/nggps_emc/scratch/Russ.Treadon/git/gsi/pr800/build
    Start 2: rtma
    Start 5: hafs_3denvar_hybens
    Start 4: hafs_4denvar_glbens
    Start 1: global_4denvar
    Start 3: rrfs_3denvar_rdasens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  723.52 sec
2/6 Test #6: global_enkf ......................   Passed  723.90 sec
3/6 Test #2: rtma .............................   Passed  1117.66 sec
4/6 Test #1: global_4denvar ...................   Passed  1745.38 sec
5/6 Test #5: hafs_3denvar_hybens ..............   Passed  1929.58 sec
6/6 Test #4: hafs_4denvar_glbens ..............   Passed  2049.51 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 2049.52 sec
```

Given the above success, please move `export FI_VERBS_PREFER_XRC=0` from `regression_var.sh` into `regression_param.sh`.

Another item:  `regression/regression_var.sh` sets `export group="ufs-ard"` for `gaeac5` and `gaeac6`.  This works for your account but it doesn't work for my account.  I don't belong to `ufs-ard`.  I changed the group lines to `export group="nggps_emc"` to get functional `noscrub` and `ptmp` paths.  Developers will need to be aware of this detail when they run ctests.

Comment (DavidBurrows-NCO) [reply regression/regression_param.sh]: I moved FI_VERBS_PREFER_XRC and re committed the change.

**Review by RussTreadon-NOAA — APPROVED** [review]
Approve.

**Comment by RussTreadon-NOAA** [comment]
@DavidBurrows-NCO , NCO said the Cactus upgrade encountered some issues which they are working through.  I'm not sure when the development WCOSS2 machine will come back online.  While we wait, I installed this PR on both Dogwood and Cactus.  I'm ready to run on when either machine becomes dev.

**Comment by DavidBurrows-NCO** [comment]
@RussTreadon-NOAA Thanks for the info, and thanks for your quick back and forth with this PR.  Have a good weekend!

**Comment by RussTreadon-NOAA** [comment]
**WCOSS2 ctests**

Install `feature/c5c6conbuild` @ 8cf64348 and `develop` @ b0e3cbaf on Cactus.  Run ctests with the following results
```
Test project /lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/pr800/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  730.56 sec
2/6 Test #6: global_enkf ......................   Passed  857.98 sec
3/6 Test #2: rtma .............................   Passed  974.22 sec
4/6 Test #5: hafs_3denvar_hybens ..............   Passed  1220.23 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  1280.17 sec
6/6 Test #1: global_4denvar ...................   Passed  1683.64 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 1683.75 sec
```

**Pull request merged**